tests/test_rf.py

```diff
@@ -198,6 +199,20 @@ def test_make_classification_hard_vote_score_mix(self):
         accuracy = rf.score(test_ds)
         self.assertGreater(accuracy, 0.7)
 
+    def test_iris(self):
+        """Tests RandomForestClassifier with a minimal example."""
+        x, y = datasets.load_iris(return_X_y=True)
+        ds_fit = load_data(x[::2], 30, y[::2])
+        ds_validate = load_data(x[1::2], 30, y[1::2])
+        rf = RandomForestClassifier(n_estimators=1, max_depth=1,
+                                    random_state=0)
+        rf.fit(ds_fit)
+        accuracy = rf.score(ds_validate)
+
+        # Accuracy should be <= 0.666... , often exactly equal.
+        self.assertLess(accuracy, 0.67)
+        self.assertGreater(accuracy, 0.34)
```

Comment:
Would it make sense to compare our accuracy to sklearn?

Reply:
The accuracy cannot be directly compared to sklearn for the same seed, because the internal logic that calls RandomState differ. I have checked manually that the results of multiple runs with different random states have a similar distribution as sklearn, but doing this analysis in the test would be costly.

I've changed the accuracy assertion to `self.assertAlmostEqual(accuracy, 2/3)`, which works for the given seed. This may cause the test to fail after a legit refactor of the code, but allows us to detect more potential bugs.